### PR TITLE
Support viewing cluster config with `tctl get cluster_config`

### PR DESCRIPTION
**Pull request by andrejtokarcik**
A first step towards resource-based dynamic configuration (#5147), this PR adds support for `tctl`-getting the `services.ClusterConfig` resource.

### `tctl get cluster_config`
```
kind: cluster_config
metadata:
  id: 1608236248009338844
  name: cluster-config
spec:
  audit: {}
  client_idle_timeout: 0s
  cluster_id: eaf63d41-a135-439b-8701-318aedd0660a
  disconnect_expired_cert: false
  keep_alive_count_max: 3
  keep_alive_interval: 5m0s
  local_auth: true
  proxy_checks_host_keys: "yes"
  session_control_timeout: 0s
  session_recording: node
version: v3
```

### `tctl get cluster_config --format=text` 
```
Cluster ID                           Session Recording Mode Local Auth 
------------------------------------ ---------------------- ---------- 
eaf63d41-a135-439b-8701-318aedd0660a node                   true       
```

### CODE REVIEW

**Review by r0mant — APPROVED** [review]

File: tool/tctl/common/collection.go [hunk 1]
```diff
@@ -477,3 +477,27 @@ func (a *appCollection) toMarshal() interface{} {
...
+	return r
+}
+
+func (c *clusterConfigCollection) writeText(w io.Writer) error {
+	t := asciitable.MakeTable([]string{
+		"Cluster ID", "Session Recording", "Proxy Checks Host Keys?",
+	})
+	for _, c := range c.configs {
+		t.AddRow([]string{
+			c.GetClusterID(), c.GetSessionRecording(), c.GetProxyChecksHostKeys(),
```
Comment (a-palchikov) [inline tool/tctl/common/collection.go:498]: I believe `ProxyChecksHostKeys` might be empty as it is only applicable to a valid session recording mode, so we might need to display it differently. Maybe `not applicable` for empty? @webvictim @russjones

**Comment by benarent** [comment]
When building CLI output for `--format=text` we should aim to make it as user friendly as possible and not show Cluster IDs for text output.. 

e.g. for getting users we do this. 
```
$ tctl get users --format=text
User
------------------
andrew
ben@goteleport.com
benarent
tmp-user
```

I've removed `proxy_checks_host_keys` as it's only really disabled for OpenSSH, and isn't recommend to be disabled. 

```
$ tctl get cluster_config --format=text
Cluster Name            Session Recording Mode.    Local Auth   
----------------------- -------------------------  -----------
aws-us-east-cluster     node                       false
```

File: tool/tctl/common/collection.go (hunk 1 above)
Comment (webvictim) [reply tool/tctl/common/collection.go:498]: Yeah, this value should only be relevant when `session_recording` is set to `proxy`.

**Comment by andrejtokarcik** [comment]
The choice of the text format columns was just a judgment call on my part, thanks for weighing in, I'll definitely adjust the output according to your suggestions.

**Comment by andrejtokarcik** [comment]
Replaced the "Proxy Checks Host Keys?" column with "Local Auth".

The situation with "Cluster Name" is more complicated though: the cluster name is not part of the `cluster_config` resource, see the `tctl get cluster_config` output shown in the original description. Now it's been my understanding that the `tctl get` subcommands should be just thin wrappers around the accessors available for the respective resource kinds. In this case however, it doesn't seem possible to fetch the cluster name from the cluster config data without an additional remote API call which would seem to go against the intended uniformity/minimality of the `tctl` resource-manipulation operations.

Finally note that `cluster_name` is a resource kind on its own. To allow the user to use `tctl get` in order to obtain the cluster name, we should instead consider adding support for `tctl get cluster_name`.

**Comment by russjones** [comment]
@andrejtokarcik Yeah, `tctl get [...]` is a thin wrapper, especially when used without the `--format` flag. I think we even tell people they can do `tctl get > file.yaml` then `tctl create -f file.yaml` in our documentation.

@klizhentas @benarent Do we we want to change `tctl get` functionality here specifically when the `--format=text` is used to transform cluster ID to cluster name?

**Comment by andrejtokarcik** [comment]
@russjones In that case I think it's best to not expose this resource at all and close this PR without merging. Once the requirements are settled on which fields of `services.ClusterConfig` should be manipulable via `tctl` then I'd develop a solution that involves those fields only, or perhaps even better, that extracts those fields into a separate resource. (As far as I can tell, the only such field so far seems to be `session_recording` anyway.)

**Pull request closed without merging**